sdk/schemaregistry/schema-registry-json/src/jsonSerializer.ts

```diff
@@ -88,22 +88,24 @@ export class JsonSerializer<MessageT = MessageContent> {
    * @throws {@link Error}
    * Thrown if the deserialization failed, e.g. because reader and writer schemas are incompatible.
    */
-  async deserialize(message: MessageT): Promise<unknown> {
+  async deserialize(message: MessageT, options?: DeserializeOptions): Promise<unknown> {
     const { data, contentType } = convertMessage(message, this.messageAdapter);
     const schemaId = getSchemaId(contentType);
-    const validator = await this.getSchemaById(schemaId);
+    const schema = await this.getSchemaById(schemaId);
     const returnedMessage = wrapError(
       () => JSON.parse(decoder.decode(data)),
       `Json deserialization failed with schema ID (${schemaId}). See 'cause' for more details.`
     );
-    wrapError(
-      () => validator(returnedMessage),
-      `Json validation failed with schema ID (${schemaId}). See 'cause' for more details.`
-    );
+    const validate = options?.validate;
+    if (validate) {
+      wrapError(
+        () => validate(returnedMessage, schema),
+        `Validation failed with schema ID (${schemaId}). See 'cause' for more details.`
```

Comment:
is `cause` going to be useful here since the validator can't actually pass any kind of string result. Maybe that should be an optional part of the return value instead of a simple boolean? something like:

```
interface ValidationResult {
   failed: boolean;
   failureReason?: string;
}
```

Reply:
`cause` is the actual error that the `validate` function has. We just wrap the error in our message and we provide the actual context where the error occurs, so I don't think we need a separate interface for the validation result.

Reply:
So the contract is the `validate` function throws when there is an error? In that case, why have it return a boolean?

Reply:
Yes, I misunderstood. The `validate` function will return void and raise an error if the validation is not successful. I will update the documentation for this. 

For more context, there are two approaches the team discusses, the `validate` method that raises the error or the `isValid` method that returns false for unsuccessful validation. We're leaning towards raising error as it is the pattern in the avro package. This might change depending on the feedback and the team discussion.